File: Gauss_Jacobi.cpp
//Gauss Jacobi and Gauss Seidel in CPP
#include<iostream>
#include<cmath>

using namespace std;
int main(){
    int n,itr;

cout<<"Enter the any value of n :";
cin>>n;

int A[n][n],b[n];
double x[n],xn[n];

cout<<"\n Enter the Matrix Elements of A : \n";
for(int i=0;i<n;i++){
    for(int j=0;j<n;j++){
        cin>>A[i][j];
    }
}

cout<<"\n Enter the b vector : \n";
for(int i=0;i<n;i++){
    cin>>b[i];
}

cout<<"\n Enter the initial values of x vector";
for(int i=0;i<n;i++){
        cin>>x[i];
}


cout<<"\n Enter the no. of iterations : ";
cin>>itr;

cout<<"\n Gauss-Jacobi Results are : \n";
for(int k=0;k<itr;k++){
    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            if(i<j){
                x[i]=(b[i]-A[i][j]*x[j])/A[i][i];
            }
            if(i>j){
                x[i]=(b[i]-A[i][j]*xn[j])/A[i][i];
            }
        }
       
    }
     for(int i=0;i<n;i++){
    xn[i]=x[i];
}
}
cout<<"x[0] = "<<x[0]<<","<<"x[1] = "<<x[1];
return 0;
}

File: Gauss_Jacobi.kt
fun main() {
     println("Enter the size of matrix")
     val n: Int = readLine()!!.toInt() //Declaring size of the matrix
     println("Enter the input : ")
     var mat = arrayOf(Array(n) { 0 }, Array(n) { 0 }) //Creating matrix

     for (i in 0..1) {
         for (j in 0..1) {
             mat[i][j] = readLine()!!.toInt()
         }
     }

     println("Enter the b vector :")
     var b = Array(n) { 0 }
     for (i in 0..1) {
         b[i] = readLine()!!.toInt()
     }


     println("Enter the initial value of x vector :")
     var x = Array<Double>(n) { 0.0 }
     for (i in 0..1) {
         x[i] = readLine()!!.toDouble()
     }
     
     var xn = Array<Double>(n) { 0.0 }
     for (i in 0..1) {
         xn[i] = x[i]
     }
     println("Enter the no. of iteration")
     var iter = readLine()!!.toInt()
     for (i in 0..iter - 1) {
         for (i in 0..1) {
             for (j in 0..1) {
                 if (i < j) {
                     x[i] = (b[i] - x[j] * mat[i][j]) / mat[i][i]
                 }
                 if (i > j) {
                     x[i] = (b[i] - xn[j] * mat[i][j]) / mat[i][i]
                 }
             }

         }
         for (i in 0..1) {
             xn[i] = x[i]
         }
     }
	 println("Gauss Jacobi : " + "x[0]=" + x[0] + "," + "x[1]=" + x[1])

 }